src/regex.c

```diff
@@ -286,19 +287,22 @@ int do_regex_hybrid_crack(struct db_main *db, const char *regex,
 	}
 
 	while (c_iterator_next(iter)) {
-		c_iterator_value(iter, buffer);
-		c_simplestring_to_utf8_string(buffer, &word[0], sizeof(word));
 		c_simplestring_clear(buffer);
-		if (options.internal_cp != UTF_8)
-			utf8_to_cp_r(word, word, sizeof(word));
+		c_iterator_value(iter, buffer);
+		word = c_simplestring_to_string(buffer);
+    /**
+      * rexgen already creates the correct encoding
+      */
+		//if (options.internal_cp != UTF_8)
+		//	utf8_to_cp_r(word, word, sizeof(word));
 		if (options.mask) {
```

Comment:
Are we handling the case of eg. --internal-encoding somewhere else? Not sure how that would happen as we never set a locale (setting a locale isn't really relevant as we may use two locales at once).